### PR TITLE
Remove generic Meter<T>

### LINKED ISSUE #4049: Remove Meter<T>
dotnet/extensions defines a Meter\<T> type that doesn't work with the newly introduced MeterFactory and doesn't align with guidance we plan to give users for writing metrics in a DI-aware scenario. We should remove this type.

### PR DESCRIPTION
**Pull request by xakep139**
This fixes #4049 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4342)

### CODE REVIEW

**Review by noahfalk — APPROVED** [review]
A few comments inline, otherwise looked good 👍

File: src/Libraries/Microsoft.AspNetCore.HeaderParsing/HeaderParsingMetrics.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        _cacheAccessCounter = Metric.CreateCacheAccessCounter(_meter);
+    }
+
+    public void CacheAccessed(string headerName, string type)
+        => _cacheAccessCounter.Add(1, headerName, type);
+
+    public void ParsingErrorOccurred(string headerName, string? kind)
+        => _parsingErrorCounter.Add(1, headerName, kind);
+
+    public void Dispose()
```
[the remark this answers is not in the document]
Comment (xakep139) [reply src/Libraries/Microsoft.AspNetCore.HeaderParsing/HeaderParsingMetrics.cs:30]: Done

File: eng/Versions.props [hunk 2]
```diff
@@ -38,6 +38,7 @@
     <MicrosoftExtensionsConfigurationVersion>8.0.0-rc.2.23464.16</MicrosoftExtensionsConfigurationVersion>
     <MicrosoftExtensionsDependencyInjectionAbstractionsVersion>8.0.0-rc.2.23464.16</MicrosoftExtensionsDependencyInjectionAbstractionsVersion>
     <MicrosoftExtensionsDependencyInjectionVersion>8.0.0-rc.2.23464.16</MicrosoftExtensionsDependencyInjectionVersion>
+    <MicrosoftExtensionsDiagnosticsVersion>8.0.0-rc.2.23464.16</MicrosoftExtensionsDiagnosticsVersion>
```
Comment (xakep139) [inline eng/Versions.props:41]: TODO: make corresponding changes in `Version.Details.xml`

**Review by tekian — APPROVED** [review]

**Review by geeknoid — APPROVED** [review]

**Review by dariusclay — APPROVED** [review]
Just housekeeping remains for build to pass, looks good

File: src/Libraries/Microsoft.Extensions.Diagnostics.Testing/Metrics/MetricCollector.cs [hunk 3]
```diff
@@ -250,7 +250,7 @@ public Task WaitForMeasurementsAsync(int minCount, CancellationToken cancellatio
     [SuppressMessage("Resilience", "EA0014:The async method doesn't support cancellation", Justification = "Not relevant in this case")]
     public async Task WaitForMeasurementsAsync(int minCount, TimeSpan timeout)
     {
-#if NET8_0_OR_GREATER
+#if NET8_0
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Diagnostics.Testing/Metrics/MetricCollector.cs:253]:
```suggestion
#if NET8_0_OR_GREATER
```

File: src/Libraries/Microsoft.Extensions.Http.Resilience/FaultInjection/Internal/HttpClientFaultInjectionMetrics.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.Metrics;
+
+namespace Microsoft.Extensions.Http.Resilience.FaultInjection.Internal;
+
+// TODO: can be also a generic Http[Client]ResilienceMetrics class
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Http.Resilience/FaultInjection/Internal/HttpClientFaultInjectionMetrics.cs:8]: @martintmk do you have any thoughts on a better name here?

**Review by iliar-turdushev — APPROVED** [review]

File: src/Libraries/Microsoft.Extensions.Resilience/FaultInjection/Internals/FaultInjectionMetrics.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.Metrics;
+
+namespace Microsoft.Extensions.Resilience.FaultInjection.Internals;
+
+// TODO: can be also a generic ResilienceMetrics class
```
[the remark this answers is not in the document]
Comment (martintmk) [reply src/Libraries/Microsoft.Extensions.Resilience/FaultInjection/Internals/FaultInjectionMetrics.cs:8]: `FaultInjection` won't be in 8.0 release so I wouldn't worry about this part too much.

File: src/Libraries/Microsoft.Extensions.Http.Resilience/FaultInjection/Internal/HttpClientFaultInjectionMetrics.cs (hunk 4 above)
Comment (martintmk) [reply src/Libraries/Microsoft.Extensions.Http.Resilience/FaultInjection/Internal/HttpClientFaultInjectionMetrics.cs:8]: I think current name is fine.

File: src/Libraries/Microsoft.Extensions.Resilience/FaultInjection/Internals/FaultInjectionMetrics.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Resilience/FaultInjection/Internals/FaultInjectionMetrics.cs:8]: Oh, I see. Please don't forget to remove the dependency on `Microsoft.Extensions.Diagnostics` when `FaultInjection` will be removed.

Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Resilience/FaultInjection/Internals/FaultInjectionMetrics.cs:8]:
```suggestion
```

**Pull request merged**